qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -45,19 +40,25 @@ def __init__(self, q_job, api, is_device):
             is_device (bool): whether backend is a real device  # TODO: remove this after Qobj
         """
         super().__init__()
-        self._q_job = q_job
-        self._qobj = q_job.qobj
+        qobj = q_job.qobj
+        self._job_data = {
+            'circuits': qobj['circuits'],
+            'hpc':  None if 'hpc' not in qobj['config'] else qobj['config']['hpc'],
+            'seed': qobj['circuits'][0]['config']['seed'],  # TODO <-- [0] ???
+            'shots': qobj['config']['shots'],
+            'max_credits': qobj['config']['max_credits']
+        }
         self._api = api
-        self._job_id = None  # this must be before creating the future
-        self._backend_name = self._qobj.get('config').get('backend_name')
+        self._job_id = None
+        self._backend_name = qobj.get('config').get('backend_name')
         self._status = JobStatus.INITIALIZING
-        self._future_submit = self._executor.submit(self._submit)
         self._status_msg = None
         self._cancelled = False
-        self._exception = None
         self._is_device = is_device
-        self._from_api = False
-        self.creation_date = None
+        self._queue_position = 0
+        self._creation_date = None
+        self._executor = futures.ThreadPoolExecutor()
```

Comment:
I think the reason I made the executor a class level variable was because I observed bad async behavior. If it's instantiated in a job instance then it only ever handles one process!

Reply:
Ok, I haven't noticed such a behavior but I don't want to deal with async nature problems in Python (no way). I don't mind having it as a class level variable, even though the executor is actually handling just one task indeed. As stated in other comment, we have to review our Executor strategy.

Reply:
With the executor as a class level variable it parallelizes all tasks for that class. Some async issues came up in another issue; they were trying to do another async on top of this one and it caused python to hang. If this really is an issue perhaps we could have a qiskit level executor instances which everyone uses.